pkg/kv/kvserver/replica_proposal_quota.go

```diff
@@ -164,9 +164,12 @@ func (r *Replica) updateProposalQuotaRaftMuLocked(
 		) {
 			return
 		}
-
-		// Only consider followers that that have "healthy" RPC connections.
-		if err := r.store.cfg.NodeDialer.ConnHealth(rep.NodeID, r.connectionClass.get()); err != nil {
+		// At this point, we know that either we communicated with this replica
+		// recently, or we became the leader recently. The latter case is ambiguous
+		// w.r.t. the actual state of that replica, but it will be disambiguated
+		// soon. We do a second check using the NodeLiveness to increase the chance
+		// of disambiguating here.
+		if !r.store.cfg.NodeLiveness.IsAvailable(rep.NodeID) {
```

Comment:
`NodeLiveness` field is nil in some of the tests. Is it expected?

Reply:
Do you know the condition under which it's nil? Is it only in tests, or can happen in the real binary too?

Reply:
I mean, it can be nil very early during server startup, but I'm pretty sure that we set it up before spinning up the Raft replicas.

Reply:
Done.